content/blog/2022/2022-11-07-structured-logging-with-cloudwatch.md

```diff
@@ -0,0 +1,280 @@
+---
+authors: [pratikdas]
+title: "Structured Logging with Amazon CloudWatch"
+categories: ["AWS"]
+date: 2023-04-25 00:00:00 +1100
+excerpt: "The primary purpose of logging in applications is to debug and trace one or more root causes of unexpected behavior. However, a log without a consistent structure with context information is difficult to search for and locate the root cause of problems. This is where we need to use Structured Logging. CloudWatch is Amazon's native service for observing and monitoring resources and applications running in AWS as well as outside. In this article, we will produce structured logs from an application and ingest them in CloudWatch. We will use different search and visualization capabilities of CloudWatch to observe the behavior of our application."
+image: images/stock/0117-queue-1200x628-branded.jpg
+url: struct-log-with-cw
+---
+
+The primary purpose of logging in applications is to debug and trace one or more root causes of an unexpected behavior. We take various approaches to logging from putting ad-hoc print statements to embedding sophisticated logging libraries in our code. 
+
+Irrespective of which approach we take, a log without a consistent structure with context information is difficult to search for and locate the root cause of problems.
+
+Amazon CloudWatch is a managed monitoring and logging service which is used as centralized log storage. It can also run queries on structured logs to extract valuable information. 
+
+In this article, we will understand:
+- how to produce structured logs from applications
+- ingest structured logs in Amazon CloudWatch
+- run queries on the ingested logs to extract useful insights into the application
+
+{{% github "https://github.com/thombergs/code-examples/tree/master/aws/structured-logging-cw" %}}
+
+## What is Structured Logging 
+Before going further, let us understand Structured logging in a bit more detail. 
+
+"Structured logging is a methodology to log information in a consistent format that allows logs to be treated as data rather than text. "
```

Comment:
The sentence is in quotation marks. Does it, come from a specific source or is it a general statement?